### PR TITLE
Add scenarios API and static mocks for Policy and Agreement endpoints

### PR DESCRIPTION
**Pull request by benjaoming**
This PR includes new static mocks and simple Gherkin scenarios

Contents and tasks:

* [x] Updates schema and endpoints with some missing parts
* [x] Adds static mock for Policy and Agreement endpoints
* [x] Gherkin test for Agreement
* [x] Gherkin test for Policy
* [ ] Adds static mock for AgreementPurpose endpoint
* [ ] Adds static mock for AgreementData endpoint
* [ ] Gherkin test for AgreementData
* [ ] Gherkin test for AgreementPurpose
* [ ] Iterate over `examples/` folder, start and test each example - cc: @taylordowns2000 you mentioned this and I think that Circle CI can start parameterized jobs. I might split this change out into a separate PR since it can get a bit complex.

### CODE REVIEW

File: test/gherkin/tests/test_get_policy.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+
+@given(
+    "an Agreement for MCC Registration exists",
+    target_fixture="mcc_registration_create"
+)
+def mcc_registration_create():
+    # Currently, this is just mocked in Caddy, so we don't create the object,
+    # we just return a hard-coded ID
+    return 1
```
Comment (benjaoming) [inline test/gherkin/tests/test_get_policy.py:20]: @taylordowns2000 

Notice that this simply shouldn't be done.

Yes, we can do something in the mock application that we rely on for a test. But the other examples won't be able to follow here. We need to actually create the objects that we expect to get back in later steps.

**Review by lalc — APPROVED** [review]

File: examples/mock/static-responses/agreement-1-mcc.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+    "id": "1",
```
Comment (benjaoming) [inline examples/mock/static-responses/agreement-1-mcc.json:2]:
```suggestion
    "id": "uuid",
```

File: test/gherkin/tests/test_get_policy.py (hunk 1 above)
Comment (lalc) [reply test/gherkin/tests/test_get_policy.py:20]: This iteration its a mock.

**Comment by benjaoming** [comment]
Re: the task that I just added

> Iterate over `examples/` folder, start and test each example - cc: @taylordowns2000 you mentioned this and I think that Circle CI can start parameterized jobs. I might split this change out into a separate PR since it can get a bit complex.

@taylordowns2000 actually why should we iterate? Maybe it's fine to add an entry for each example in `.circleci/config.yml` since Building Blocks won't have more than a few and it will be manageable - and maybe we need to disable one or start it with slightly different parameters. It gives flexibility and avoids dynamic pipelines which I think might be a bit too much DevOps engineering to lift right now :)

**Comment by benjaoming** [comment]
The work in the remaining tasks are continuing. @heysatya was right to ask for the PR the be merged as-is. This was not a problem since the remaining work can easily go in after.... and....

:tada: all tests are passing :tada:

**Pull request merged**